### PR TITLE
[REVIEW]Check and invisible the button if any product is out of stock.

### PR DESCRIPTION
**Pull request by lqdung-lockon**

### CODE REVIEW

File: src/Eccube/Controller/Mypage/MypageController.php [hunk 1]
```diff
@@ -199,14 +200,26 @@ public function history(Request $request, $order_no)
...
         $this->eventDispatcher->dispatch(EccubeEvents::FRONT_MYPAGE_MYPAGE_HISTORY_INITIALIZE, $event);
 
+        /** @var Order $Order */
         $Order = $event->getArgument('Order');
 
         if (!$Order) {
             throw new NotFoundHttpException();
         }
 
+        $StockFind = true;
```
Comment (lqdung-lockon) [inline src/Eccube/Controller/Mypage/MypageController.php]: $StockFind => $stockFind

File: src/Eccube/Controller/Mypage/MypageController.php [hunk 2]
```diff
@@ -199,14 +200,26 @@ public function history(Request $request, $order_no)
...
+        foreach ($Order->getShippings() as $shipping) {
```
Comment (lqdung-lockon) [inline src/Eccube/Controller/Mypage/MypageController.php]: dùng $Order->getOrderItems(); is ok

File: src/Eccube/Controller/Mypage/MypageController.php [hunk 3]
```diff
@@ -199,14 +200,26 @@ public function history(Request $request, $order_no)
...
+                if ($orderItem->isProduct() && $orderItem->getProductClass()->getStockFind() == false) {
+                    $StockFind = false;
+                    break;
+                }
+            }
+        }
+
         return [
             'Order' => $Order,
+            'StockFind' => $StockFind
```
Comment (lqdung-lockon) [inline src/Eccube/Controller/Mypage/MypageController.php]: 'StockFind' chuyển thành 'stock_find'

File: src/Eccube/Resource/template/default/Mypage/history.twig [hunk 4]
```diff
@@ -180,8 +180,11 @@ file that was distributed with this source code.
                     </dl>
                     {% endif %}
                     <div class="ec-totalBox__total">{{'common.label.cart_total'|trans}}<span class="ec-totalBox__price">{{ Order.payment_total|price }}</span><span class="ec-totalBox__taxLabel">{{'shopping.label.tax_incl'|trans}}</span></div>
+                    {% if StockFind %}
                     <a href="{{ url('mypage_order', {'order_no': Order.order_no }) }}" class="ec-blockBtn--action load-overlay" {{ csrf_token_for_anchor() }} data-method="put" data-confirm="false">{{'mypage.history.detail.re_order'|trans}}</a>
-
+                    {% else %}
+                        <button class="ec-blockBtn--action" disabled="disabled">ただいま品切れ中です</button>
```
Comment (lqdung-lockon) [inline src/Eccube/Resource/template/default/Mypage/history.twig]: translate please

**Comment by hoand-vn** [comment]
finish fixing

**Pull request closed without merging**